Block/Analytics.php

```diff
@@ -62,7 +62,10 @@ public function getEvent()
                 return $this->categoryViewEvent->callJS();
             // catalog search pages
             case 'catalogsearch_result_index':
-                return $this->catalogSearchEvent->callJS();
+                return $this->catalogSearchEvent->callJS($this->actionContext->getRequest()->getParam('q'));
+            // catalog advanced result page
+            case 'catalogsearch_advanced_result':
+                return $this->catalogSearchEvent->callJS($this->actionContext->getRequest()->getParam('name'));
```

Comment:
This is leaking the logic outside of the helpers we specifically created for each event.

Reply:
Moved the param extraction inside the search helper class.